source/server/server.cc

```diff
@@ -56,9 +56,14 @@ InstanceImpl::InstanceImpl(const Options& options, Event::TimeSystem& time_syste
                            ThreadLocal::Instance& tls, Thread::ThreadFactory& thread_factory,
                            Filesystem::Instance& file_system,
                            std::unique_ptr<ProcessContext> process_context)
-    : workers_started_(false), shutdown_(false), options_(options), time_source_(time_system),
-      restarter_(restarter), start_time_(time(nullptr)), original_start_time_(start_time_),
-      stats_store_(store), thread_local_(tls),
+    : workers_started_(false), shutdown_(false), options_(options),
+      validation_context_(
```

Comment:
AFAICT this code, members, etc. are duplicated between the validation server and the real server. Perhaps it could all be bundled together in a ProdValidationContext or something like that? Up to you.